File: problems/205-isomorphic-strings/python/205_isomorphic_strings.py
from typing import Dict


class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        if s == t:
            return True
        return self.toNumeric(s) == self.toNumeric(t)

    def toNumeric(self, s: str) -> str:
        dict: Dict[chr, int] = {}
        n: int = 1
        numeric: s = ""
        for c in s:
            if c in dict.keys():
                numeric += str(dict[c])
            else:
                dict[c] = n
                numeric += str(n)
                n += 1
        return numeric


def test(s: str, t: str) -> None:
    result: bool = Solution().isIsomorphic(s, t)
    print(f"{s} -> {t}: {result}")


if __name__ == "__main__":
    test("egg", "add")
    test("foo", "bar")
    test("paper", "title")

File: problems/205-isomorphic-strings/csharp/205_isomorphic_strings.cs
using System.Collections.Generic;
using System;

public static class Program
{
  public static void Main(string[] args)
  {
    Solution s = new Solution();
    Console.WriteLine(s.IsIsomorphic("egg", "add"));
    Console.WriteLine(s.IsIsomorphic("foo", "bar"));
    Console.WriteLine(s.IsIsomorphic("paper", "title"));
  }
}

public class Solution
{
  public bool IsIsomorphic(string s, string t)
  {
    if (s.Length != t.Length) return false;
    if (s.Equals(t)) return true;
    return ToNumeric(s).Equals(ToNumeric(t));
  }
  public string ToNumeric(string str)
  {
    Dictionary<char, int> dict = new Dictionary<char, int>();
    int n = 1;
    string numeric = string.Empty;
    foreach (char c in str)
      if (dict.ContainsKey(c))
        numeric += dict[c];
      else
      {
        dict.Add(c, n);
        numeric += n.ToString();
        n++;
      }
    return numeric;
  }
}